src/crbmg.c

```diff
@@ -1,52 +1,55 @@
 /** @file
-    @author ATOR @date 2005-11-29
-*/
-
-
+ *  @brief Read the next message from a BUFR file that was
+ *  previously opened for reading via a C language interface.
+ */
 #include "bufrlib.h"
 #include "cobfl.h"
 
-/**
-C
-C SUBPROGRAM:    CRBMG
-C   PRGMMR: ATOR             ORG: NP12       DATE: 2005-11-29
-C
-C ABSTRACT:  THIS ROUTINE READS THE NEXT BUFR MESSAGE FROM THE SYSTEM
-C   FILE MOST RECENTLY OPENED FOR READING/INPUT VIA BUFR ARCHIVE LIBRARY
-C   ROUTINE COBFL.  ANY BUFR EDITION 0 OR EDITION 1 MESSAGES THAT ARE
-C   READ ARE AUTOMATICALLY CONVERTED TO BUFR EDITION 2.
-C
-C PROGRAM HISTORY LOG:
-C 2005-11-29  J. ATOR    -- ORIGINAL AUTHOR
-C
-C USAGE:    CALL CRBMG( BMG, MXMB, NMB, IRET )
-C   INPUT ARGUMENT LIST:
-C     MXMB     - INTEGER: DIMENSIONED SIZE (IN BYTES) OF BMG; USED
-C                BY THE ROUTINE TO ENSURE THAT IT DOES NOT OVERFLOW
-C                THE BMG ARRAY
-C
-C   OUTPUT ARGUMENT LIST:
-C     BMG      - CHARACTER*1: ARRAY CONTAINING BUFR MESSAGE
-C     NMB      - INTEGER: SIZE (IN BYTES) OF BUFR MESSAGE IN BMG
-C     IRET     - INTEGER: RETURN CODE:
-C                  0 = normal return
-C                  1 = overflow of BMG array
-C                  2 = "7777" indicator not found in expected location 
-C                 -1 = end-of-file encountered while reading
-C                 -2 = I/O error encountered while reading
-C
-C REMARKS:
-C    THIS ROUTINE CALLS:        BORT     GETS1LOC ICHKSTR  IPKM
-C                               IUPBS01  IUPM     RBYTES
-C    THIS ROUTINE IS CALLED BY: None
-C                               Normally called only by application
-C                               programs.
-C
-C ATTRIBUTES:
-C   LANGUAGE: C
-C   MACHINE:  PORTABLE TO ALL PLATFORMS
-C
-C$$$*/
+/** 
+ *  This subroutine reads the next BUFR message from the system
+ *  file that was opened via the most recent call to subroutine
+ *  cobfl() with io = 'r'.
+ *  
+ *  @author J. Ator
+ *  @date 2005-11-29
+ *
+ *  @param[in] mxmb     - integer: Dimensioned size (in bytes) of
+ *                        BMG; used by the subroutine to ensure that
+ *                        it doesn't overflow the BMG array
+ *  @param[out] bmg     - character*(*): array containing BUFR
+ *                        message
+ *  @param[out] nmb     - integer: size (in bytes) of BUFR message
+ *                        in BMG
+ *  @param[out] iret    - integer: return code
+ *                         - 0 = normal return
+ *                         - 1 = overflow of BMG array
+ *                         - 2 = "7777" indicator not found in
+ *                               expected location 
+ *                         - -1 = end-of-file encountered while
+ *                               reading
+ *                         - -2 = I/O error encountered while reading
+ *
+ * <p>This subroutine is designed to be easily callable from
```

Comment:
I don't think you need the <p> tags.

Reply:
I've found in some cases that if I don't include the `<p>` tags, then all of the text discussion gets mashed up together in one big block.  So to be safe and to keep the output more readable, I've been including them, but only where I've found from experience that they're needed.